Expensify/expensify-common
# Update simply-deferred library to use Expensify fork

@marcaaron @AndrewGable will you please review this?
cc @tgolen

This PR is to update usages of `simply-deferred` for any package that uses expensify-common. We are updating ReactNativeChat to use the newly forked [Expensify simply-deferred](https://github.com/Expensify/simply-deferred) so that `npm ci` can run successfully, and as a result we'll have to update that library and reference everywhere else it's used. Currently, this needs to be updated in the following draft PRs:

Web: https://github.com/Expensify/Web-Expensify/pull/29850
Onyx: https://github.com/Expensify/react-native-onyx/pull/18
ReactNativeChat: https://github.com/Expensify/ReactNativeChat/pull/918
Mobile: Will make PR after this is merged

Once this PR is merged, I will update the above PRs to use the merge hash here, and get those in review.

### Fixed Issues
$ Related to https://github.com/Expensify/Expensify/issues/144368

# Tests
I tested this on both Web-Expensify and ReactNativeChat repos using the latest hash here. Once this is merged, I'll update Web, Mobile, Onyx, and RNC.

0. For context, the latest hash of our simply-deferred lib is here: https://github.com/Expensify/simply-deferred/commits/master
1. Check out the branch for the Web-PR: https://github.com/Expensify/Web-Expensify/pull/29850
2. On web, run `npm install`, confirm there isn't a diff
3. Confirm you can create expenses, reports, and chat via concierge
4. For RNC testing, I confirmed this is working by running `npm ci` on the GH push action and confirming the original issue no longer happens
    1. Before the fix, the script failed at [this line](https://github.com/Expensify/ReactNativeChat/runs/1520804799#step:7:458) which is due to the old dependencies syntax
    2. From the latest test after updating the dependencies, you can see that the error that the desktop build fails at is related to the GH access token [here](https://github.com/Expensify/ReactNativeChat/runs/1534860860#step:7:470), which is after the build
5. Checkout the RNC PR listed above, run `npm install`, run the RNC app, confirm there are no issues that you can't reproduce on master

# QA
None
